crates/biome_service/src/workspace.rs

```diff
@@ -984,3 +998,35 @@ fn test_order() {
         assert!(items[0] < items[1], "{} < {}", items[0], items[1]);
     }
 }
+
+new_key_type! {
+    pub struct WorkspaceKey;
+}
+
+#[derive(Debug, Default)]
+pub struct WorkspaceData<T> {
+    /// [DenseSlotMap] is the slowest type in insertion/removal, but the fastest in iteration
+    ///
+    /// Users wouldn't change workspace folders very often,
+    pub paths: DenseSlotMap<WorkspaceKey, T>,
+}
+
+impl<T> WorkspaceData<T> {
```

Comment:
Why do we add these methods if we commonly access `paths` directly anyway? I would choose between either adding methods and keeping `paths` private, or making `paths` public and removing these methods.